lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2523,6 +2535,7 @@ def print_pdf(self, filename, **kwargs):
                 bbox_inches_restore=_bbox_inches_restore)
             self.figure.draw(renderer)
             renderer.finalize()
+            file.finalize()
```

Comment:
The `file.close()` call in the `finally` section no longer does the finalization. The finalization is the part that cannot be reasonably done if something has raised an exception (in this case, the missing font) so it belongs in the `try` section.